### PR TITLE
add OpenMP offloading example

### PR DESCRIPTION
**Pull request by matthiasdiener**
*Original date: 2019-04-01 18:51:26*
*Original PR: https://charm.cs.illinois.edu/gerrit/4947*

---

Change-Id: Ib2686dbef8b111717b09081a839fae9879465fba

### CODE REVIEW

File: examples/charm++/shared_runtimes/openmp-offload/vecadd/hello.C [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#include <stdio.h>
+#include "hello.decl.h"
+#include <omp.h>
+#include <typeinfo>
+
+/* readonly */ CProxy_Main mainProxy;
+
+class Main : public CBase_Main {
+public:
+  Main(CkArgMsg* m) {
```
Comment (stwhite91) [inline examples/charm++/shared_runtimes/openmp-offload/vecadd/hello.C]: m is leaked

Comment (matthiasdiener) [reply examples/charm++/shared_runtimes/openmp-offload/vecadd/hello.C]: fixed.

**Comment by matthiasdiener** [comment]
This is ready for review.

File: examples/charm++/shared_runtimes/openmp-offload/vecadd/Makefile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+vecadd.decl.h: vecadd.ci
+	$(CHARMC) vecadd.ci
+
+clean:
+	rm -f *.decl.h *.def.h *.o vecadd charmrun
+
+vecadd.o: vecadd.C vecadd.decl.h
+	$(CHARMC) $(OMP_OPT) -c vecadd.C
+
+test: all
```
Comment (stwhite91) [inline examples/charm++/shared_runtimes/openmp-offload/vecadd/Makefile]: Will this example build and run on all autobuild targets? Or should it not run on certain builds?

Comment (matthiasdiener) [reply examples/charm++/shared_runtimes/openmp-offload/vecadd/Makefile]: The `shared_runtimes` directory isn't executed by `make test` so it shouldn't run on the test builds.

**Pull request merged**